src/action.cpp

```diff
@@ -1559,7 +1559,17 @@ void actionUpdateDroid(DROID *psDroid)
 							// building a gun tower over a wall - OK
 							if (droidStartBuild(psDroid))
 							{
-								objTrace(psDroid->id, "DACTION_MOVETOBUILD: start building");
+								objTrace(psDroid->id, "DACTION_MOVETOBUILD: start building defense");
+								psDroid->action = DACTION_BUILD;
+							}
+						}
+						else if ((psStruct->pStructureType->type == REF_FACTORY && order->psStats->type == REF_FACTORY_MODULE) ||
+								(psStruct->pStructureType->type == REF_RESEARCH && order->psStats->type == REF_RESEARCH_MODULE) ||
+								(psStruct->pStructureType->type == REF_POWER_GEN && order->psStats->type == REF_POWER_MODULE)) {
+							// upgrade current structure in a row
+							if (droidStartBuild(psDroid))
```

Comment:
What about checking for REF_VTOL_FACTORY too? Otherwise it'll break with VTOL factories I presume.

I find the brace here on this else-if to be rather awkward so if you could add the brace on its own line that would look better (which we prefer as our "standard").

Reply:
yup, you re right, added that too